src/library_tty.js

```diff
@@ -63,13 +63,22 @@ mergeInto(LibraryManager.library, {
           if (result === null || result === undefined) break;
           bytesRead++;
           buffer[offset+i] = result;
+
+          // If EOL character, return having put it in the buffer.
+          if (result === 10) break;
         }
         if (bytesRead) {
           stream.node.timestamp = Date.now();
         }
         return bytesRead;
       },
       write: function(stream, buffer, offset, length, pos) {
+#if USE_PRINTCHARS_FOR_STDOUT == 1
```

Comment:
In general we have had an idiom to just do `#if USE_PRINTCHARS_FOR_STDOUT`  rather than `#if USE_PRINTCHARS_FOR_STDOUT == 1` in scenarios where the intent is not to explicitly stress the value of `1`.